### ISSUE TITLE
Lost constant

### ISSUE THREAD

**Issue by rfranke** [OpenModelica/OpenModelica#7660]
The function `Modelica.Math.Matrices.LAPACK.dgees` defines
```
pure function dgees
  ...
protected
  constant Integer dummyFunctionPointerNotUsed[1]={0};
  ...
external"FORTRAN 77" dgees(
          "V",
          "N",
          dummyFunctionPointerNotUsed,
          n,
          T,
          lda,
          sdim,
          eval_real,
          eval_imag,
          Z,
          lda,
          work,
          lwork,
          bwork,
          info) annotation (Library={"lapack"});
  ...
end dgees;
```

The compilation fails, claiming that `dummyFunctionPointerNotUsed` would be undeclared. See [ModelicaTest.Math.TestMatrices2](https://libraries.openmodelica.org/branches/master/ModelicaTest_trunk/files/ModelicaTest_trunk_ModelicaTest.Math.TestMatrices2.err)

```
ModelicaTest_trunk_ModelicaTest.Math.TestMatrices2_functions.c\nModelicaTest_trunk_ModelicaTest.Math.TestMatrices2_functions.c:3075:96:
 error: use of undeclared identifier '_dummyFunctionPointerNotUsed_ext'
  dgees_((char*)MMC_STRINGDATA(tmp4), (char*)MMC_STRINGDATA(tmp5), data_of_integer_f77_array(&(_dummyFunctionPointerNotUsed_ext)), (int*) &_n, data_of_real_f77_array(&(_T_ext)), (int*) &_lda, (int*) &_sdim, data_of_real_f77_array(&(_eval_real_ext)), data_of_real_f77_array(&(_eval_imag_ext)), data_of_real_f77_array(&(_Z_ext)), (int*) &_lda, data_of_real_f77_array(&(_work_ext)), (int*) &_lwork, data_of_boolean_f77_array(&(_bwork_ext)), (int*) &_info_ext);
 ```

**Comment by adrpo** [comment]
Looks like the replacement of constants is not working in the external C declaration. 
The frontend generates the correct flat Modelica but it seems the backend doesn't know how to handle it properly.

**Comment by mahge** [comment]
The issue is related to filtering of _input_, _output_, and _local_ (a.k.a _bidir_) variables of a function. The intention was to filter out and separate "variables" as in **elements**. 

The filtering function used here, however, also checks the `VARIABILITY` attribute of the variable/element and expects `VARIABILITY==VARIABLE` so it skips all `CONSTANT `variables. 

I will push a fix shortly.

**Comment by mahge** [comment]
This immediate issue is fixed by #7666. 

However, the models do not yet compile due to undefined references to a function. See #7667. 

I will close this ticket since the reported problem is fixed and we can continue in the other ticket.
